.github/workflows/go.yml

```diff
@@ -9,40 +9,97 @@ on:
       - master
 
 jobs:
-
-  test:
-    name: Build + Test
-    runs-on: ubuntu-18.04
+  terraform-fmt:
+    name: Terraform fmt
+    runs-on: ubuntu-20.04
     steps:
-
-      - name: Set up Go 1.15
-        uses: actions/setup-go@v1
-        with:
-          go-version: 1.15
-
       - name: Check out code into the Go module directory
         uses: actions/checkout@v2
 
       - name: Execute terraform fmt -check -recursive .
         run: terraform fmt -check -recursive .
 
+  go-fmt:
+    name: Go fmt
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
       - name: Execute gofmt -l .
         run: gofmt -l .
 
       - name: Execute test -z $(gofmt -l .)
         run: test -z $(gofmt -l .)
 
+  build:
+    name: Go build
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.15
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
       - name: Get dependencies
         run: go get -v -t -d ./...
 
       - name: Build
         run: go build -v ./...
 
+  go-vet:
+    name: Go vet
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.15
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Get dependencies
+        run: go get -v -t -d ./...
+
       - name: Static code analysis
         run: go vet ./...
 
+  go-test:
+    name: Go test
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.15
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Get dependencies
+        run: go get -v -t -d ./...
+
       - name: Test
         run: go test ./...
 
+  go-test-integration:
+    name: Go test integration
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.15
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Get dependencies
+        run: go get -v -t -d ./...
+
       - name: Integration tests
-        run: go test -p 1 --tags='integration,fixtures' ./test/...
+        run: go test -p 3 --tags='integration,fixtures' ./test/...
```

Comment:
Why 3? Maybe we can rely on GOMAXPROCS?

Reply:
i tried, but then in fact the github runner is so busy that it can't finish the tests at all